packages/cli/src/reporters/ci.ts

```diff
@@ -0,0 +1,35 @@
+import { RunCompletion, Dataset } from "@empiricalrun/types";
+import { markdownSummary, failedOutputsSummary } from "../stats";
+import { promises as fs } from "fs";
+
+async function prepareOutputForPRComment(
+  completion: RunCompletion[],
+  dataset: Dataset,
+) {
+  if (process.env.GITHUB_OUTPUT) {
+    const failedOutputs = failedOutputsSummary(completion);
+    let failedSummary = "";
+
+    if (failedOutputs) {
+      const { code, message } = failedOutputs;
+      failedSummary = `**Error**: Some outputs were not generated successfully\n${code}: ${message}`;
+    }
+
+    const datasetLength = `Total dataset samples: ${dataset.samples?.length || 0}`;
+    // Multiline output
+    // https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+    await fs.appendFile(
+      process.env.GITHUB_OUTPUT,
+      `result<<EOF\n${markdownSummary(completion)}\n${datasetLength}\n\n${failedSummary}\nEOF`,
+    );
+  }
+}
+
+export async function reportOnCI(
+  completion: RunCompletion[],
+  dataset: Dataset,
+) {
+  if (process.env.GITHUB_ACTIONS === "true") {
```

Comment:
can this check and the github output check be at a single place ?

Reply:
hmm, yes, can do that. the idea was to:
1. check for "which ci provider" in this function 
2. check for output type in the github action function (e.g. we can do a different method there if we don't want to send a PR comment)